### PR TITLE
feat: Slack huddle → KMS importer (dual-write + dedup-gate routing)

### PR DESCRIPTION
**Pull request by ryaker**
## Summary

Adds an importer that turns Slack huddle AI-notes (the canvas Slackbot
posts after a huddle ends) into KMS entries via the dual-write pattern,
mirroring PR #74's Granola architecture.

- ONE whole-huddle `memory` entry per huddle (distilled summary, not raw canvas markdown), tagged with `metadata.subject="Slack.huddle.<channel>.<date>"` and `metadata.source_doc="slack://canvas/<team_id>/<file_id>"`.
- 2–5 typed claim entries per huddle (OB1 6-type taxonomy → KMSmcp `contentType`), each linking back via `metadata.related_to=[<summary_id>]` and sharing the same `source_doc` pointer.
- Both write paths route through `unified_store` so the dedup gate fires naturally — overlap with Granola transcripts of the same call gets caught upstream (`action=complement` or duplicate).

## Architecture

Source abstraction (mirrors PR #74's `GranolaSource`):

- **`LiveSlackSource`** — dependency-injected `slack_search_public` + `slack_read_canvas` callbacks. Production path; runs from a Claude Code session that has the Slack MCP authorized. Exposed via `runImportLive()` for in-session callers.
- **`FileSlackSource`** — reads pre-dumped JSON for cron/launchd contexts. Accepts both a pre-resolved-array shape AND a `{ messages, canvases }` shape so the same script handles fast cron paths and live agent dumps.

Distillation: Claude Haiku 4.5 (`claude-haiku-4-5-20251001`) via `@anthropic-ai/sdk` (already in `package.json`).

Sync log: `~/.kms-slack-huddle-sync.json` keyed by `canvas_file_id` for idempotent re-runs. Failed huddles are NOT added to the log so they retry on next run.

## Files

All in `src/scripts/` (PR #75's pattern, for ts-jest + `import.meta` compat):

- `src/scripts/slack-huddle-distill-prompt.ts` — Haiku 4.5 prompt, JSON validator, OB1→contentType map, canvas-id regex, Slackbot recap detector.
- `src/scripts/import-slack-huddles.ts` — CLI driver, `MinimalMcpClient` (StreamableHTTP), OAuth client-credentials fallback, `runImport`/`processHuddle`/`runImportLive` entrypoints.
- `src/scripts/import-slack-huddles-cli.ts` — thin CLI shim (avoids top-level `import.meta` so the main module stays unit-testable under ts-jest CJS loader).
- `src/__tests__/import-slack-huddles.test.ts` — 66 tests.
- `.gitignore` — added `!src/scripts/` (the bare `scripts/` rule was globbing to any `scripts/` directory).

## Sample CLI invocation

```bash
# Cron path: agent dumps huddles to JSON, then this script imports.
KMS_BEARER_TOKEN=eyJ... \
  node dist/scripts/import-slack-huddles-cli.js \
  --input ~/slack-huddles-dump.json \
  --workspace tengo

# Smoke test (no Anthropic, no KMS writes — walks the pipeline):
node dist/scripts/import-slack-huddles-cli.js --input ~/slack-huddles-dump.json --dry-run

# Live path (from a Claude Code session via runImportLive()):
import { parseArgs, runImportLive } from './src/scripts/import-slack-huddles.ts'
await runImportLive({
  opts: parseArgs(['--source', 'live', '--bearer-token', '...']),
  searchPublic: (q, o) => mcp__Slack__slack_search_public({ query: q, limit: o.limit, sort: o.sort }),
  readCanvas:   (id) => mcp__Slack__slack_read_canvas({ canvas_id: id })
})
```

The `messages+canvases` JSON shape lets a one-shot agent dump matter
once and let cron handle the import:

```json
{
  "messages": [
    { "user": "USLACKBOT", "ts": "...", "text": "AI huddle notes are ready... <https://x.slack.com/docs/T0/F1|...>", "channel": { "id": "C1", "name": "core-values" } }
  ],
  "canvases": { "F1": "# Core Values huddle\n..." }
}
```

## Test plan

- [x] `npx tsc --noEmit` clean
- [x] `npm run build` clean
- [x] `npx jest` — 145/145 pass (was 79 baseline; +66 new)
- [x] CLI `--help` renders
- [x] CLI `--dry-run --input <file>` walks pipeline end-to-end with two sample huddles
- [x] CLI rejects `--source bogus` and unknown flags
- [x] CLI exits 2 when `--input` is missing in file mode
- [ ] **Manual production verification** — once merged, run a real dump from `projcaribou.slack.com` and confirm the dedup gate catches Granola overlap

## What was NOT touched

Per the constraint list, none of these in-flight PR surfaces were modified:
`SparrowDBStorage.ts`, `Mem0Storage.ts`, `EmbeddingService.ts`, `UnifiedStoreTool.ts`. `@anthropic-ai/sdk` was already a dependency from PR #74 — no version bump.

## Notes / ambiguity resolutions

1. **`learning → insight` (not `fact`).** PR #74 maps `learning → fact`, but PR #75 maps `learning → insight`. Slack huddles are conversational/decisional artifacts more akin to PR #75's curated documents than to PR #74's passive-observation transcripts, so I used PR #75's mapping. Easy to flip if desired.
2. **`qualitative_confidence` over `confidence`.** Aligned with PR #75's field name (vs PR #74's `confidence`) since the importer lives in `src/scripts/` like the md-corpus importer. Cross-importer schema consistency.
3. **Channel slug uses `_` as a word char.** `slugifyChannel('OPS_Daily Sync!')` → `ops_daily-sync` (not `ops-daily-sync`). Slack already separates words with `-`, so I preserved underscores rather than collapse them.
4. **Subject falls back to `unknown-date` when `huddle_date` missing.** Better than mid-build crash; corrected on re-import once the dump fixes the date.
5. **`--source live` from the bare CLI exits with a helpful message.** It can't reach the Slack MCP from a plain Node process; the `runImportLive()` programmatic entry is the seam for in-session callers.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/scripts/slack-huddle-distill-prompt.ts [hunk 1]
```diff
@@ -0,0 +1,276 @@
...
+ */
+export function extractCanvasId(
+  messageText: string
+): { teamId: string; fileId: string } | null {
+  if (typeof messageText !== 'string' || messageText.length === 0) return null
+  // Match the docs/<team_id>/<file_id> path. Tolerate either a bare URL or
+  // mrkdwn `<url|label>` wrapping. Team IDs and file IDs in Slack are
+  // alphanumeric and start with T / F respectively, but we accept any
+  // [A-Z0-9]+ to stay forward-compatible.
+  const re = /slack\.com\/docs\/([A-Z0-9]+)\/([A-Z0-9]+)/
```
[the remark this answers is not in the document]
Comment (ryaker) [reply src/scripts/slack-huddle-distill-prompt.ts]: Fixed — changed the regex to `/slack\.com\/docs\/([a-zA-Z0-9]+)\/([a-zA-Z0-9]+)/i`. The `/i` flag makes the whole match case-insensitive; the explicit `[a-zA-Z0-9]+` class also handles any mixed-case IDs that might appear in edge cases. All existing extractCanvasId tests still pass.

File: src/scripts/import-slack-huddles.ts [hunk 2]
```diff
@@ -0,0 +1,1125 @@
...
+    captureSession: boolean
+  ): Promise<{ sessionId: string | null; body: any }> {
+    const headers: Record<string, string> = {
+      'Content-Type': 'application/json',
+      'Accept': 'application/json, text/event-stream'
+    }
+    if (this.bearer) headers['Authorization'] = `Bearer ${this.bearer}`
+    if (this.sessionId) headers['mcp-session-id'] = this.sessionId
+
+    const res = await fetch(this.kmsUrl, {
```
[the remark this answers is not in the document]
Comment (ryaker) [reply src/scripts/import-slack-huddles.ts]: Fixed — added a 30 s `AbortController` timeout to the `postRaw` fetch (same pattern used in `EmbeddingService.ts`). The timer is cleared in a `finally` block to avoid a hanging timer if the request completes normally. The `close()` DELETE call is best-effort and already swallows errors, so no timeout was added there.

File: src/scripts/import-slack-huddles.ts [hunk 3]
```diff
@@ -0,0 +1,1125 @@
...
+  }
+}
+
+function parseSseToJson(sse: string): any {
+  const lines = sse.split(/\r?\n/)
+  const dataLines = lines.filter(l => l.startsWith('data: ')).map(l => l.slice(6))
+  if (dataLines.length === 0) {
+    throw new Error(`Could not parse MCP response (not JSON, not SSE): ${sse.slice(0, 200)}`)
+  }
+  const joined = dataLines.join('')
```
[the remark this answers is not in the document]
Comment (ryaker) [reply src/scripts/import-slack-huddles.ts:495]: Fixed — `parseSseToJson` now splits the body into per-event blocks (separated by blank lines) rather than joining all `data:` lines into a single string. Each block's payload is tried as JSON; the last successfully-parsed block is returned (the result event, which always arrives after any notification events). A fallback to the original join is retained for single-event bodies where the data field itself spans multiple lines. This handles the case Gemini flagged — multiple discrete SSE events — without breaking the single-event path.
